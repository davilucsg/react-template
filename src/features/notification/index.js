export {default as notificationContainer } from './views/notification.container';