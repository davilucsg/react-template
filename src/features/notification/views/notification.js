import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import CloseIcon from '@material-ui/icons/Close';
import green from '@material-ui/core/colors/green';
import amber from '@material-ui/core/colors/amber';
import IconButton from '@material-ui/core/IconButton';
import Snackbar from '@material-ui/core/Snackbar';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import { withStyles } from '@material-ui/core/styles';

export const notification = props => {
  const { classes, className, mensagem, variant, exibenotification, ocultanotification } = props;
  return (
    <Snackbar
      id="snackBarnotification"
      anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
      open={exibenotification}
      autoHideDuration={4000}
      onClose={ocultanotification}
      style={{ display: mensagem.length ? 'block' : 'none' }}
    >
      {mensagem.length ? (
        <SnackbarContent
          id="snackBackContent"
          className={classNames(classes[variant], className)}
          aria-describedby="client-snackbar"
          message={mensagem}
          action={[
            <IconButton
              id="iconOcultarnotification"
              key="close"
              aria-label="Close"
              color="inherit"
              onClick={ocultanotification}
            >
              <CloseIcon />
            </IconButton>
          ]}
        />
      ) : null}
    </Snackbar>
  );
};

notification.defaultProps = {
  variante: 'success',
  mensagem: ''
};

notification.propTypes = {
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  mensagem: PropTypes.node || PropTypes.string,
  onClose: PropTypes.func,
  variant: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  ocultanotification: PropTypes.func
};

const styles = theme => ({
  success: {
    backgroundColor: green[600]
  },
  error: {
    backgroundColor: theme.palette.error.dark
  },
  info: {
    backgroundColor: theme.palette.primary.dark
  },
  warning: {
    backgroundColor: amber[700]
  },
  icon: {
    fontSize: 20
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: theme.spacing.unit
  },
  message: {
    display: 'flex',
    alignItems: 'center'
  }
});

export default withStyles(styles)(notification);
