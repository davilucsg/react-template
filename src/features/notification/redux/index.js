import * as notificationActions from './notification.actions';
export { default as notificationReducers } from './notification.reducers';
export { notificationActions };
