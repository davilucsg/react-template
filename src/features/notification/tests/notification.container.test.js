import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import { shallow,configure } from 'enzyme';
import { NotificationContainer } from '../views/notification.container';
import  Notification  from '../views/notification';
configure({ adapter: new Adapter() });

describe('Testa <notificationContainer />', () => {
  let component;
  beforeEach(() => {
    component = shallow(<NotificationContainer reduxProps={{}}/>);
  });
  it('Deve renderizar <notificationContainer /> ', () => {
    expect(component).toMatchSnapshot();
  });
  it('Deve renderizar <notification /> ', () => {
    expect(component.find(Notification).exists()).toEqual(true);
  });

  it('Deve setar props exibenotification para true e state exibenotification deve ser true ', () => {
    component.setProps({
      reduxProps: { exibenotification: true }
    });
    expect(component.state().exibenotification).toEqual(true);
  });

  it('Deve setar props exibenotification para false e state exibenotification deve ser false ', () => {
    component.setProps({
      reduxProps: { exibenotification: false }
    });
    expect(component.state().exibenotification).toEqual(false);
  });
});
