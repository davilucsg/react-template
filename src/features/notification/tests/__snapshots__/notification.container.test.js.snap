// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa <notificationContainer /> Deve renderizar <notificationContainer />  1`] = `
<Fragment>
  <WithStyles(notification) />
</Fragment>
`;
